src/ui/control/attribution_control.js

```diff
@@ -186,6 +174,17 @@ class AttributionControl {
             return true;
         });
 
+        if (this.options.customAttribution) {
+            if (Array.isArray(this.options.customAttribution)) {
+                const customAttributionReversed = [...this.options.customAttribution].reverse();
+                for (const attribution of customAttributionReversed) {
+                    if (typeof attribution === 'string') attributions.unshift(attribution);
+                }
```

Comment:
Nit: slicing, reversing and unshifting gets a little confusing — maybe worth simplifying?

```suggestion
                const attributions = [...attributions, ...this.options.customAttribution.filter(s => typeof s === 'string')];
```